### ISSUE TITLE
Alternative S3-compatible targets

### ISSUE THREAD

**Issue by bsideup** [myniva/gradle-s3-build-cache#8]
Hi! 

Could you please provide an option to configure a custom endpoint?

We would like to use this plugin with DigitalOcean's Spaces (fully S3-compatible storage)

Thanks!

**Comment by myniva** [comment]
Hi!

Thanks for your interest in this plugin and for suggesting an improvement to it.

Unfortunatelly, I don't see an easy way to add this. We are using AWS official Java SDK and as far as I could see there's no way to confiugre a custom endpoint there.

If you see an option how this could be implemented, please let us know.

**Comment by bsideup** [comment]
Hi @myniva,

You can use custom endpoints with official AWS SDK by doing something like this:
```java
AmazonS3ClientBuilder s3ClientBuilder = AmazonS3ClientBuilder.standard();

if (endpoint != null) {
    s3ClientBuilder.setEndpointConfiguration(
            new AwsClientBuilder.EndpointConfiguration(
                    endpoint,
                    region
            )
    );
}
AmazonS3 amazonS3 = s3ClientBuilder.build();
```

**Comment by myniva** [comment]
Cool, thanks for pointing that out! Will implement the option to configure the endpoint ASAP.

**Comment by myniva** [comment]
Implemented in release 0.5.0
